### PR TITLE
feat: index compound harvests for vault, change api resp source to ddb

### PR DESCRIPTION
**Pull request by NamelsKing**

### CODE REVIEW

File: src/vaults/interfaces/vault-harvests-model.interface.ts [hunk 1]
```diff
@@ -25,7 +25,10 @@ export class VaultHarvestsModel implements VaultHarvestsExtended {
 
   @Title('amount')
   @Description('amount of harvested token')
-  @Example(15.34534)
+  @Example({
```
Comment (axejintao) [inline src/vaults/interfaces/vault-harvests-model.interface.ts:28]: looking back at this, we should format this to human readable balances as `number`

File: src/vaults/models/harvest-compound.model.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+@table(HARVEST_COMPOUND_DATA)
+export class HarvestCompoundData implements VaultHarvestsExtended {
+  @hashKey()
+  vault!: string;
+
+  @rangeKey({ defaultProvider: () => Date.now() })
+  timestamp!: number;
+
+  @attribute({ memberType: embed(BigNumberDynamodbModel) })
+  amount!: BigNumber;
```
Comment (axejintao) [inline src/vaults/models/harvest-compound.model.ts:18]: let's use number

File: src/vaults/vaults.utils.spec.ts [hunk 3]
```diff
@@ -43,11 +44,17 @@ import { ONE_DAY_SECONDS, ONE_YEAR_MS } from '../config/constants';
...
+import * as tokenUtils from '../tokens/tokens.utils';
+import { vaultsHarvestsSdkMock } from './mocks/vaults-harvests-sdk.mock';
+import * as gqlGenT from '@badger-dao/sdk/lib/graphql/generated/badger';
+import { vaultsGraphSdkMapMock } from './mocks/vaults-graph-sdk-map.mock';
 
 describe('vaults.utils', () => {
   const vault = getVaultDefinition(TEST_CHAIN, TOKENS.BBADGER);
 
   beforeEach(() => {
+    // console.log = jest.fn();
```
Comment (axejintao) [inline src/vaults/vaults.utils.spec.ts:56]: can remove

**Review by axejintao — COMMENTED** [review]
lgtm, just the format comment

File: src/indexers/harvest-compound-indexer.spec.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    jest.spyOn(BadgerSDK.prototype, 'ready').mockImplementation();
+    jest.spyOn(Chain.prototype, 'getSdk').mockImplementation();
+  });
+
+  describe('indexVaultsHarvestsCompund', () => {
+    it('should save all harvests from all chains to ddb', async () => {
+      await indexVaultsHarvestsCompund();
+
+      expect(put.mock.calls.length).toBe(22);
```
[the remark this answers is not in the document]
Comment (NamelsKing) [reply src/indexers/harvest-compound-indexer.spec.ts:38]: ye, but as i know exact amount of mocked harvests, i can be sure of amount of `put` calls

File: src/vaults/interfaces/vault-harvests-model.interface.ts (hunk 1 above)
Comment (NamelsKing) [reply src/vaults/interfaces/vault-harvests-model.interface.ts:28]: i'll need to shake all call chain, but makes sense

File: src/vaults/vaults.utils.spec.ts (hunk 3 above)
Comment (NamelsKing) [reply src/vaults/vaults.utils.spec.ts:56]: sure

File: src/vaults/models/harvest-compound.model.ts (hunk 2 above)
Comment (NamelsKing) [reply src/vaults/models/harvest-compound.model.ts:18]: ok, lol, number cause ddb field to overflow, so i've used string

File: src/vaults/__snapshots__/vaults.utils.spec.ts.snap [hunk 5]
```diff
@@ -1,5 +1,154 @@
...
+    "estimatedApr": 0,
+    "eventType": "Harvest",
+    "strategyBalance": 2255.6396126895474,
+    "timestamp": 1640731025,
+    "token": "0x075b1bb99792c9e1041ba13afef80c91a1e70fb3",
+  },
+  Object {
+    "amount": 84.17266705379542,
+    "block": 13612910,
+    "estimatedApr": Infinity,
```
Comment (axejintao) [inline src/vaults/__snapshots__/vaults.utils.spec.ts.snap:71]: 🤔

Comment (NamelsKing) [reply src/vaults/__snapshots__/vaults.utils.spec.ts.snap:71]: Aaaaaaaaaaa

**Review by axejintao — APPROVED** [review]

**Pull request merged**